### DISCUSSION TITLE
Foundry Fundamentals / Section 2 / Video 22

### DISCUSSION THREAD

**Discussion by AlimGlck** [Cyfrin/foundry-full-course-cu#2692] [General]
If you have reached this video, you may know that it is kind of problematic for editing and also explaining things. I went through the codes in that video to understand better and to try to figure out some chunks of codes that we used that I didn't understand why we used. Then I realized some of them were really redundant, indeed. Because when I omitted those code lines, the test was successful anyway. So I decided to share those lines because maybe even if tests pass without those codes there is something that they do useful in some scenarios that I haven't discovered.

That code chunks are

```
 FundMe(payable(mostRecentlyDeployed)).fund{value:SEND_VALUE}();
 ```
 
 When I omitted the ```payable``` it worked perfectly fine.

  ```
   function run() external{

        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment("Fund Me",block.chainid);
        
        fundFundMe(mostRecentlyDeployed);
        
    }
    ```
    and
    
    ```
    function run() external{

        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment("Fund Me",block.chainid);
        
        withdrawFundMe(mostRecentlyDeployed);
        
    }
    ```
    
    

If anybody there here to inform us about those code chunks, please do.

**Comment by EngrPips** [comment]
If you omitted the `payable` keyword here `FundMe(payable(mostRecentlyDeployed)).fund{value:SEND_VALUE}();` and it still works then i think you mark it as payable somewhere before actually sending ether to it.

The below code gets the address of the latest deployed `FundMe` contract and pass it to the `fundFundMe` script which will actually fund it.

```js
function run() external{

      address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment("Fund Me",block.chainid);
      
      fundFundMe(mostRecentlyDeployed);
      
  }
```

The below code retrieved the address of the latest deployed `FundMe` contract and passed it to the `withdrawFundMe` which will withdraw from it.
```js
 function run() external{

      address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment("Fund Me",block.chainid);
      
      withdrawFundMe(mostRecentlyDeployed);
      
  }
```
